dev/lint-python

```diff
@@ -96,6 +98,32 @@ fi
 
 rm "$PEP8_REPORT_PATH"
 
+# Check that the documentation builds acceptably, skip check if sphinx is not installed.
+if hash "$SPHINXBUILD" 2> /dev/null; then
+  cd python/docs
+  make clean
+  export SPHINXOPTS=""
+  make linkcheck &>> "$SPHINX_REPORT_PATH"  || lint_status=1
```

Comment:
Should be `&>` instead of `&>>`. Otherwise I got the following error message:

~~~
dev/lint-python: line 106: syntax error near unexpected token `>'
dev/lint-python: line 106: `  make linkcheck &>> "$SPHINX_REPORT_PATH"  || lint_status=1'
~~~

Btw, do we really want to verify links? Links may die due to external website changes. If a PR fails because of this, the author may not even know how to fix them.